lib/std/fs/file.zig

```diff
@@ -414,10 +414,12 @@ pub const File = struct {
     pub fn read(self: File, buffer: []u8) ReadError!usize {
         if (is_windows) {
             return windows.ReadFile(self.handle, buffer, null, self.intended_io_mode);
-        } else if (self.capable_io_mode != self.intended_io_mode) {
-            return std.event.Loop.instance.?.read(self.handle, buffer);
-        } else {
+        }
+
+        if (self.intended_io_mode == .blocking or !std.io.is_async) {
```

Comment:
```suggestion
        if (self.intended_io_mode == .blocking) {
```

`or !std.io.is_async` is dead code because when blocking I/O is being used, the type of intended_io_mode is io.ModeOverride which is `enum { blocking }` (a 0-bit type) which makes `self.intended_io_mode == .blocking` comptime true.

```zig
/// This is an enum value to use for I/O mode at runtime, since it takes up zero bytes at runtime,
/// and makes expressions comptime-known when `is_async` is `false`.
pub const ModeOverride = if (is_async) Mode else enum { blocking };
```